cmd/thv-operator/pkg/registryapi/config/config.go

```diff
@@ -485,3 +560,167 @@ func buildDatabaseConfig(dbConfig *mcpv1alpha1.MCPRegistryDatabaseConfig) *Datab
+		if err != nil {
+			return nil, fmt.Errorf("failed to build OAuth provider configuration: %w", err)
+		}
+		config.Providers = append(config.Providers, *provider)
+	}
+
+	return config, nil
+}
+
+// buildOAuthProviderConfig creates an OAuthProviderConfig from the CRD spec.
+func buildOAuthProviderConfig(
+	providerSpec *mcpv1alpha1.MCPRegistryOAuthProviderConfig,
+) (*OAuthProviderConfig, error) {
+	if providerSpec == nil {
+		return nil, fmt.Errorf("provider specification is required")
+	}
+
+	if providerSpec.Name == "" {
+		return nil, fmt.Errorf("provider name is required")
+	}
+	if providerSpec.IssuerURL == "" {
+		return nil, fmt.Errorf("provider issuer URL is required")
+	}
+	if providerSpec.Audience == "" {
+		return nil, fmt.Errorf("provider audience is required")
+	}
+
+	config := &OAuthProviderConfig{
+		Name:           providerSpec.Name,
+		IssuerURL:      providerSpec.IssuerURL,
+		Audience:       providerSpec.Audience,
+		AllowPrivateIP: providerSpec.AllowPrivateIP,
+	}
+
+	// JwksUrl is optional - if specified, OIDC discovery is skipped
+	if providerSpec.JwksUrl != "" {
+		config.JwksUrl = providerSpec.JwksUrl
+	}
+
+	// ClientID is optional, so we only set it if provided
+	if providerSpec.ClientID != "" {
+		config.ClientID = providerSpec.ClientID
+	}
+
+	// IntrospectionURL is optional - used for validating opaque tokens
+	if providerSpec.IntrospectionURL != "" {
+		config.IntrospectionURL = providerSpec.IntrospectionURL
+	}
+
+	// For ClientSecretRef, CACertRef, and AuthTokenRef, we store the path where the secret/configmap
+	// will be mounted. The actual mounting is handled by the pod template spec builder.
+	// The registry server will read the file at runtime.
+	if providerSpec.ClientSecretRef != nil {
+		// Secret will be mounted at /secrets/{secretName}/{key}
+		config.ClientSecretFile = buildSecretFilePath(providerSpec.ClientSecretRef)
+	}
+
+	// CaCertPath can be set directly or via CACertRef
+	// Direct path takes precedence over reference
+	if providerSpec.CaCertPath != "" {
+		config.CACertPath = providerSpec.CaCertPath
+	} else if providerSpec.CACertRef != nil {
+		// ConfigMap will be mounted at /config/certs/{configMapName}/{key}
+		config.CACertPath = buildCACertFilePath(providerSpec.CACertRef)
+	}
+
+	// AuthTokenFile can be set directly or via AuthTokenRef
+	// Direct path takes precedence over reference
+	if providerSpec.AuthTokenFile != "" {
+		config.AuthTokenFile = providerSpec.AuthTokenFile
+	} else if providerSpec.AuthTokenRef != nil {
+		// Secret will be mounted at /secrets/{secretName}/{key}
+		config.AuthTokenFile = buildSecretFilePath(providerSpec.AuthTokenRef)
+	}
+
+	return config, nil
+}
+
+// buildSecretFilePath constructs the file path where a secret will be mounted
+func buildSecretFilePath(secretRef *corev1.SecretKeySelector) string {
+	if secretRef == nil {
+		return ""
+	}
+	key := secretRef.Key
+	if key == "" {
+		key = "clientSecret"
```

Comment:
the code uses a hardcoded default key `clientSecret` when no key is specified, but this function is used for both `ClientSecretRef` and `AuthTokenRef`, for the latter a different default might be better (such as `token`).

Currently, if a user provides authTokenRef with just name: my-token-secret (no key), the resulting path will be /secrets/my-token-secret/clientSecret